components/onboarding-modal/OnboardingModal.js

```diff
@@ -0,0 +1,233 @@
+
+import Modal, { ModalBody, ModalHeader, ModalFooter } from '../../components/StyledModal';
+import OnboardingNavButtons from './OnboardingNavButtons';
+import OnboardingStepsProgress from './OnboardingStepsProgress';
+import OnboardingContentBox from './OnboardingContentBox';
+import MessageBox from '../../components/MessageBox';
+
+import { getErrorFromGraphqlException } from '../../lib/utils';
+import { getLoggedInUserQuery } from '../../lib/graphql/queries';
+import { Router } from '../../server/pages';
+
+const StepsProgressBox = styled(Box)`
+  min-height: 95px;
+  max-width: 600px;
+
+  @media screen and (max-width: 640px) {
+    width: 100%;
+    max-width: 100%;
+  }
+`;
+
+class OnboardingModal extends React.Component {
+  static propTypes = {
+    query: PropTypes.object,
+    collective: PropTypes.object,
+    LoggedInUser: PropTypes.object,
+    // refetchLoggedInUser: PropTypes.func,
+    EditCollectiveMembers: PropTypes.func,
+    EditCollectiveContact: PropTypes.func,
+    show: PropTypes.bool,
+    setShow: PropTypes.func,
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      step: 0,
+      members: [],
+      error: null,
+    };
+  }
+
+  componentDidMount() {
+    this.setStep(this.props.query.step);
+  }
+
+  componentDidUpdate(oldProps) {
+    if (oldProps.query.step !== this.props.query.step) {
+      this.setStep(this.props.query.step);
+    }
+  }
+
+  setStep = queryStep => {
+    if (queryStep === undefined) {
+      this.setState({ step: 0 });
+    } else if (queryStep === 'administrators') {
+      this.setState({ step: 1 });
+    } else if (queryStep === 'contact') {
+      this.setState({ step: 2 });
+    }
+  };
+
+  addAdmins = members => {
+    this.setState({ members });
+  };
+
+  addContact = (name, value) => {
+    this.setState(state => ({
+      collective: {
+        ...state.collective,
+        [name]: value,
+      },
+    }));
+  };
+
+  submitAdmins = async () => {
+    try {
+      this.setState({ isSubmitting: true });
+      await this.props.EditCollectiveMembers({
+        collectiveId: this.props.collective.id,
+        members: this.state.members.map(member => ({
+          id: member.id,
+          role: member.role,
+          member: {
+            id: member.member.id,
+            name: member.member.name,
+          },
+        })),
+      });
+    } catch (e) {
+      this.setState({ isSubmitting: false, error: getErrorFromGraphqlException(e) });
+    }
+  };
+
+  submitContact = async () => {
+    const collective = {
+      ...this.state.collective,
+      id: this.props.collective.id,
+    };
+    try {
+      this.setState({ isSubmitting: true });
+      await this.props.EditCollectiveContact({
+        collective,
+      });
+    } catch (e) {
+      this.setState({ isSubmitting: false, error: getErrorFromGraphqlException(e) });
+    }
+  };
+
+  submitCollectiveInfo = async () => {
+    await this.submitContact();
+    await this.submitAdmins();
```

Comment:
**[:hammer_and_wrench: **Important**]** Because both of these functions are protected by a try/catch, we will still redirect event if the two functions fail. Is this what we want?

---

<details>
<summary>More details about this feedback</summary>
<br/>

This feedback categorization is inspired by [Netlify's Feedback Ladders system](https://www.netlify.com/blog/2020/03/05/feedback-ladders-how-we-encode-code-reviews-at-netlify). It includes 5 levels of feedback:

:rotating_light: **Critical blocker**
This feedback blocks all related work, and requires immediate action. 

:no_entry_sign: **Blocker**
This feedback should block the work from moving forward or being approved, but it does not necessarily require immediate action from the team.

:hammer_and_wrench: **Important**
This feedback should not block the work from moving forward or being approved, but it does require future action from the team. This generally means the feedback is not required for the MVP or initial draft of this work, but should be addressed at some point and we should ideally create an issue to track it.

:sparkles: **Enhancement**
This feedback should not block the work from moving forward or being approved. The project lead or owner should consider the feedback and implement or open an issue it if they agree with it.

:ice_cream: **Details & personal preferences**
This feedback should not block the work from moving forward or being approved. The project lead or owner should consider the feedback and implement or open an issue it if they deem it useful. This is also the category for everything that relates to personal preferences or code style.

</details>

Reply:
This is a great catch - I'll be sure to keep this in mind for the next PR!